components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -206,7 +206,7 @@ object typing {
   case class TypedClass private[typing] (klass: Class[_], params: List[TypingResult]) extends SingleTypingResult {
     override val valueOpt: None.type = None
 
-    override def withoutValue: TypedClass = this
+    override def withoutValue: TypedClass = TypedClass(klass, params.map(_.withoutValue))
```

Comment:
Thank for the comment because I forgot to move test for conversion from a Set to a List and yes it's the bug. The below test will fail without upper change:
```
  test("should convert a set to a list") {
    val parsed = parse[Any]("#setVal.toList()", ctx).validValue
    parsed.returnType shouldBe Typed.genericTypeClass[JList[_]](List(Typed.typedClass[String]))
    parsed.evaluateSync[Any](ctx) shouldBe JList.of("a")
  }
```
And in `typeFunction` I do:
```
invocationTarget.withoutValue match {
      case TypedClass(klass, params) if klass.isAOrChildOf(collectionClass) || klass.isArray =>
        Typed.genericTypeClass[JList[_]](params).validNel
      case Unknown => Typed.genericTypeClass[JList[_]](List(Unknown)).validNel
      case _       => GenericFunctionTypingError.ArgumentTypeError.invalidNel
    }
```
So `invocationTarget.withoutValue` does not propagate to params.